src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -379,16 +387,16 @@ Status CoreWorkerMemoryStore::GetImpl(const std::vector<ObjectID> &object_ids,
   // calls. If timeout_ms == -1, this should run forever until all objects are
   // ready or a signal is received. Else it should run repeatedly until that timeout
   // is reached.
-  while (!timed_out && signal_status.ok() &&
-         !(done = get_request->Wait(iteration_timeout))) {
-    if (check_signals_) {
-      signal_status = check_signals_();
-    }
-
-    if (remaining_timeout >= 0) {
-      remaining_timeout -= iteration_timeout;
-      iteration_timeout = std::min(remaining_timeout, iteration_timeout);
-      timed_out = remaining_timeout <= 0;
+  {
+    std::signal(SIGINT, SignalHandler);
+    std::signal(SIGTERM, SignalHandler);
```

Comment:
Are you sure you want to overwrite default SIGTERM handler? Which means exception thrown will not have any effect

Reply:
Trying to replicate the behavior of the check_signals mentioned in the pr description, I think sigterm maps to the SystemExit is SIGTERM will double check

Reply:
hmm possibly not, removed sigterm and only keep sigint for now, we really just need to check for ctrl+c looking at the previous pr that introduced check_signals